server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndexer.java

```diff
@@ -93,29 +102,42 @@ private long doIndex(BulkIndexer bulk, Iterator<IssueDoc> issues) {
   }
 
   public void deleteProject(String uuid, boolean refresh) {
-    BulkIndexer bulk = new BulkIndexer(esClient, IssueIndexDefinition.INDEX);
+    BulkIndexer bulk = new BulkIndexer(esClient, INDEX);
     bulk.setDisableRefresh(!refresh);
     bulk.start();
-    SearchRequestBuilder search = esClient.prepareSearch(IssueIndexDefinition.INDEX)
+    SearchRequestBuilder search = esClient.prepareSearch(INDEX)
       .setRouting(uuid)
       .setQuery(QueryBuilders.filteredQuery(
         QueryBuilders.matchAllQuery(),
-        FilterBuilders.boolFilter().must(FilterBuilders.termsFilter(IssueIndexDefinition.FIELD_ISSUE_PROJECT_UUID, uuid))
+        FilterBuilders.boolFilter().must(FilterBuilders.termsFilter(FIELD_ISSUE_PROJECT_UUID, uuid))
         ));
     bulk.addDeletion(search);
     bulk.stop();
   }
 
+  public void deleteByKeys(List<String> issueKeys){
+    int count = 0;
+    BulkRequestBuilder builder = esClient.prepareBulk().setRefresh(true);
```

Comment:
refresh is an heavy operation. It must not been done between each execution of bulk requests.